### PR TITLE
[ray][metric][feature] Add metric for ray object store

### PR DESCRIPTION
**Pull request by Catch-Bull**
## What do these changes do?

Add metrics for ray object store so that we can monitor the status of ray object store

## Related issue number

Fixes #xxxx

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

File: mars/utils.py [hunk 1]
```diff
@@ -1544,3 +1546,79 @@ def __str__(self):
...
+    percentile_list = [
+        _percentile_builder[percentile_type](callback, window)
+        for percentile_type, callback, window in percentile_args
+    ]
+    st_time = time.time()
+
+    try:
+        yield
+    except Exception as error:
+        raise error
```
Comment (wjsi) [inline mars/utils.py:1620]: Is this line for debug only? If so, remove it.

File: mars/utils.py [hunk 2]
```diff
@@ -1544,3 +1546,79 @@ def __str__(self):
...
+    Percentile.PercentileType.P99: Percentile.build_p99,
+    Percentile.PercentileType.P95: Percentile.build_p95,
+    Percentile.PercentileType.P90: Percentile.build_p90,
+}
+
+
+@contextmanager
+def cost_time_percentile_record(
+    percentile_args: List[
+        Tuple[Percentile.PercentileType, Callable[[float], None], int]
```
Comment (wjsi) [inline mars/utils.py:1608]: Better to use a `typing.NamedTuple` instead.

File: mars/utils.py [hunk 3]
```diff
@@ -1544,3 +1546,79 @@ def __str__(self):
...
+
+_percentile_builder = {
+    Percentile.PercentileType.P99: Percentile.build_p99,
+    Percentile.PercentileType.P95: Percentile.build_p95,
+    Percentile.PercentileType.P90: Percentile.build_p90,
+}
+
+
+@contextmanager
+def cost_time_percentile_record(
```
Comment (wjsi) [inline mars/utils.py:1606]: Rename to `record_time_cost_percentile`?

File: mars/utils.py [hunk 4]
```diff
@@ -54,6 +55,7 @@
 import numpy as np
 import pandas as pd
 from sklearn.base import BaseEstimator
+from enum import Enum
```
Comment (wjsi) [inline mars/utils.py:58]: Put imports of built-in packages together. Move below `from contextlib import contextmanager`.

File: mars/utils.py [hunk 5]
```diff
@@ -1544,3 +1546,81 @@ def __str__(self):
...
+}
+
+PercentileArg = NamedTuple(
+    "PercentileArg",
+    [
+        ("percentile_type", Percentile.PercentileType),
+        ("callback", Callable[[float], None]),
+        ("window", int),
+    ],
+)
```
Comment (wjsi) [inline mars/utils.py:1611]:
```suggestion
class PercentileArg(NamedTuple):
    percentile_type: Percentile.PercentileType
    callback: Callable[[float], None]
    window: int
```

File: mars/utils.py [hunk 6]
```diff
@@ -1544,3 +1546,81 @@ def __str__(self):
             "__module__": bases[-1].__module__,
         },
     )().with_traceback(traceback)
+
+
+class Percentile:
```
Comment (zhongchun) [inline mars/utils.py:1551]: `Percentile` maybe moved to `mars/metrics/api.py` because it is used to record metrics.

Comment (wjsi) [reply mars/utils.py:1551]: Agree.

Comment (Catch-Bull) [reply mars/utils.py]: done.

File: mars/utils.py (hunk 1 above)
Comment (Catch-Bull) [reply mars/utils.py]: done.

File: mars/utils.py (hunk 2 above)
Comment (Catch-Bull) [reply mars/utils.py]: done.

File: mars/tests/test_utils.py [hunk 7]
```diff
@@ -586,3 +588,48 @@ def __call__(self, *args, **kwargs):
 
     func = Func()
     assert get_func_token_values(func) == [func]
+
+
+def test_percentile_report():
```
Comment (wjsi) [inline mars/tests/test_utils.py:593]: Move these tests into `mars.metrics` also.

File: mars/utils.py (hunk 3 above)
Comment (Catch-Bull) [reply mars/utils.py]: done.

File: mars/utils.py (hunk 4 above)
Comment (Catch-Bull) [reply mars/utils.py]: done.

File: mars/utils.py (hunk 5 above)
Comment (Catch-Bull) [reply mars/utils.py]: done.

File: mars/tests/test_utils.py (hunk 7 above)
Comment (Catch-Bull) [reply mars/tests/test_utils.py]: sure!

**Review by wjsi — APPROVED** [review]
LGTM
